lib/harvesting/gazetteer.ex

```diff
@@ -63,19 +76,43 @@ defmodule Argos.Harvesting.Gazetteer do
   defmodule GazetteerHarvester do
     @batch_size 100
 
+    @doc """
+    Loads data from gazetteer and saves it into the database
+    """
     def harvest!(%Date{} = lastModified) do
       query = build_query_string(lastModified)
       total = harvest_batch!(query, @batch_size)
       total
     end
 
+    def harvest!(%{placeid: _pid} = place) do
+      query = build_query_string(place)
+      response = GazetteerClient.fetch!(query)
+      save_resources!(response)
+      response["total"]
+    end
+
+    @doc """
+    Loads one document and returns it, instead of saving
+    """
+    def request!(%{placeid: _pid} = place) do
+      query = build_query_string(place)
+      %{"result" => response} = GazetteerClient.fetch!(query)
+      response
+    end
+
```

Comment:
Ist das nicht eine Dopplung mit `harvest!`?

Ah, das eine ist die Funktion die zum ES hinzufügt, das andere die die einfach den Datensatz zurückgibt? Dann würde ich es aber so bauen dass die, die zum ES hinzufügt die mit dem Datensatz anfragt und dann entsprechend in ES schiebt.

Oder vielleicht sogar noch einen Schritt zurück: Modularer wird Funktionaler Code meist dadurch, dass sich die einzelnen Funktionen nicht in Kette gegenseitig aufrufen, sondern indem es eine Hauptfunktion gibt, die dann die verschiedenen Subfunktionen nacheinander aufruft und auf die Art dafür sorgt, dass das Ergebnis nach und nach so umgeformt wird, wie sie es benötigt.

Reply:
hab die funktion gelöscht